plugins/endpoints/src/main/java/org/pytorch/serve/plugins/endpoint/Token.java

```diff
@@ -0,0 +1,219 @@
+    public String parseQuery(Request req) {
+        QueryStringDecoder decoder = new QueryStringDecoder(req.getRequestURI());
+        Map<String, List<String>> parameters = decoder.parameters();
+        List<String> values = parameters.get("type");
+        if (values != null && !values.isEmpty()) {
+            if ("management".equals(values.get(0)) || "inference".equals(values.get(0))) {
+                return values.get(0);
+            } else {
+                return "WRONG TYPE";
+            }
+        }
+        return "NO TYPE PROVIDED";
+    }
+
+    public String generateKey() {
+        byte[] randomBytes = new byte[6];
+        secureRandom.nextBytes(randomBytes);
+        return baseEncoder.encodeToString(randomBytes);
+    }
+
+    public Instant generateTokenExpiration() {
+        return Instant.now().plusSeconds(TimeUnit.MINUTES.toSeconds(timeToExpirationMinutes));
+    }
+
+    // generates a key file with new keys depending on the parameter provided
+    public boolean generateKeyFile(String type) throws IOException {
+        String userDirectory = System.getProperty("user.dir") + "/key_file.json";
+        File file = new File(userDirectory);
+        if (!file.createNewFile() && !file.exists()) {
+            return false;
+        }
+        if (apiKey == null) {
+            apiKey = generateKey();
+        }
+        switch (type) {
+            case "management":
+                managementKey = generateKey();
+                managementExpirationTimeMinutes = generateTokenExpiration();
+                break;
+            case "inference":
+                inferenceKey = generateKey();
+                inferenceExpirationTimeMinutes = generateTokenExpiration();
+                break;
+            default:
+                managementKey = generateKey();
+                inferenceKey = generateKey();
+                inferenceExpirationTimeMinutes = generateTokenExpiration();
+                managementExpirationTimeMinutes = generateTokenExpiration();
+        }
+
+        JsonArray jsonArray = new JsonArray();
+        jsonArray.add(
+                "Management Key: "
+                        + managementKey
+                        + " --- Expiration time: "
+                        + managementExpirationTimeMinutes);
+        jsonArray.add(
+                "Inference Key: "
+                        + inferenceKey
+                        + " --- Expiration time: "
+                        + inferenceExpirationTimeMinutes);
+        jsonArray.add("API Key: " + apiKey);
+
+        Files.write(
+                Paths.get("key_file.json"),
+                new GsonBuilder()
+                        .setPrettyPrinting()
+                        .create()
+                        .toJson(jsonArray)
+                        .getBytes(StandardCharsets.UTF_8));
+
+        if (!setFilePermissions()) {
+            try {
+                Files.delete(Paths.get("key_file.txt"));
```

Comment:
Filename has changed. Better to store the name in a constant somewhere to provide central edit access.

Reply:
Noted, but will still be needed to be changed in multiple places. Will make a variable for the plugin but the frontend will not use it.